### ISSUE TITLE
Refactor getCredentials for arbitrary instance filters

### ISSUE THREAD

**Issue by rwd** [germanattanasio/vcap-services#24]
I would like to propose a refactor of `getCredentials` to accept an `Object` as its only parameter, in order to:
1. make service name optional, and if omitted scan all services
2. support arbitrary instance filters
3. alleviate the need to pass `null` for unneeded instance filters

This would however be a backwards-incompatible change to the function signature. Alternatively it could be implemented as a new function, e.g. `findCredentials`.

I could provide a PR if there is interest.

## Examples

Specify service name and instance filters:
```js
var vcapServices = require('vcap_services');
var credentials = vcapServices.getCredentials({
  service: 'compose-for-redis',
  instance: { tags: 'cache' }
});
```

Specify service name only:
```js
var vcapServices = require('vcap_services');
var credentials = vcapServices.getCredentials({
  service: 'compose-for-redis'
});
```

Specify instance filters only:
```js
var vcapServices = require('vcap_services');
var credentials = vcapServices.getCredentials({
  instance: { name: 'my-redis', plan: 'Standard' }
});
```

**Comment by germanattanasio** [comment]
Related to #24.

I would prefer to have a new method called `findCredentials()`

**Comment by germanattanasio** [comment]
closed by #23
